awstesting/behaviorassert.go

```diff
@@ -0,0 +1,303 @@
+// header key in the expected map is present in the request with equal values. request may
+// have additional headers outside the expected ones.
+func AssertRequestHeadersMatch(t *testing.T, expectHeader map[string]interface{}, req *request.Request, msgAndArgs ...interface{}) bool {
+	for key, valExpect := range expectHeader {
+		valReq := req.HTTPRequest.Header.Get(key)
+		if key == "Header-Json-Value" {
+			expectJSONValue, err1 := protocol.DecodeJSONValue(valExpect.(string), protocol.Base64Escape)
+			if err1 != nil {
+				t.Fatalf("unable to parse expected JSON, %v", err1)
+			}
+			responseJSONValue, err2 := protocol.DecodeJSONValue(valReq, protocol.Base64Escape)
+			if err2 != nil {
+				t.Fatalf("unable to parse response JSON, %v", err2)
+			}
+
+			var r DiffReporter
+			for key1, val1 := range expectJSONValue {
+				if !cmp.Equal(val1, responseJSONValue[key1], FloatIntEquate(), cmp.Reporter(&r)) {
+					t.Fatalf("aws.JSON value from expect and response don't match\n%s\n", r.String())
+				}
+			}
+			continue
+		}
+		if valReq == "" || valReq != valExpect {
+			t.Fatalf("header values don't match for key: %q,\nexpect: %s\nactual: %s", key, valExpect, valReq)
+		}
+	}
+	return true
+}
+
+// AssertRequestBodyEqualsBytes asserts if base64 encoded string inside request body is equal
+// to expected value.
+func AssertRequestBodyEqualsBytes(t *testing.T, expectVal string, req *request.Request) bool {
+	bytesReqBody := ReadBody(t, req)
+	return StringEqual(t, expectVal, string(bytesReqBody)) == nil
+}
+
+// AssertRequestBodyEqualsJSON verifies that the json value in request body
+// string matches the expectVal map
+func AssertRequestBodyEqualsJSON(t *testing.T, expectVal map[string]interface{}, req *request.Request, msgAndArgs ...interface{}) bool {
+	bytesReqBody := ReadBody(t, req)
+	actualVal := map[string]interface{}{}
+	if err := json.Unmarshal(bytesReqBody, &actualVal); err != nil {
+		t.Fatalf("unable to parse expected JSON, %v", err)
+	}
+	var r DiffReporter
+	for key, val := range expectVal {
+		if key == "JsonValue" && !AssertJSON(t, reflect.ValueOf(val).String(), reflect.ValueOf(actualVal[key]).String()) {
+			t.Fatalf("AssertJSON failed when key is %s,\nexpect: %s\n\t\tactual: %s\t", key, val, actualVal[key])
+		} else if key != "JsonValue" {
+			if !cmp.Equal(val, actualVal[key], FloatIntEquate(), cmp.Reporter(&r)) {
+				t.Fatalf(r.String())
+			}
+		}
+	}
+	return true
+}
+
+// AssertRequestBodyMatchesXML verifies that the expect xml string matches the actual. True if
+// XML string inside request and expect are equal. For XML string, order of
+// elements and attributes are significant while whitespaces are not.
+func AssertRequestBodyMatchesXML(t *testing.T, expectVal string, req *request.Request, container interface{}, msgAndArgs ...interface{}) bool {
+	r := req.HTTPRequest
+	if r.Body == nil {
+		t.Fatalf("request body is nil")
+	}
+	body := util.SortXML(r.Body)
```

Comment:
This isn't applicable as of now because `AssertRequestBodyMatchesXML` uses `AssertXML` function which doesn't work.

PR# 2804 fixes `AssertXML` and `AssertRequestBodyMatchesXML` can be changed later once the PR is merged